### PR TITLE
Refactor `AccountConnection` with new layout and type components

### LINKED ISSUE #7341: [Layout foundations] Rebuild `AccountConnection` with new layout primitives

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Fixes #7341

### CODE REVIEW

File: polaris-react/src/components/AccountConnection/AccountConnection.tsx [hunk 1]
```diff
@@ -54,39 +54,37 @@ export function AccountConnection({
...
-      <TextStyle variation="subdued">{details}</TextStyle>
-    </div>
+    <Text as="span" variant="bodyMd" color="subdued">
+      {details}
+    </Text>
   ) : null;
 
   const termsOfServiceMarkup = termsOfService ? (
-    <div className={styles.TermsOfService}>{termsOfService}</div>
+    <Box paddingTop="5">{termsOfService}</Box>
```
Comment (SeanyB) [inline polaris-react/src/components/AccountConnection/AccountConnection.tsx:69]: Shouldn't we be using tokens, rather than hardcoded numbers?

Comment (SeanyB) [reply polaris-react/src/components/AccountConnection/AccountConnection.tsx:69]: Oh, wait. I see. This is a number from the set made available in the Box component props. 🤦

Comment (aveline) [reply polaris-react/src/components/AccountConnection/AccountConnection.tsx:69]: Yes it maps to our spacing tokens. But perhaps it's not very clear? I wonder if there's anything we can do to make that easier?

**Review by alex-page — APPROVED** [review]

**Review by laurkim — APPROVED** [review]
Nice!! Changes on Storybook lgtm 🚀

**Pull request merged**